include/ultra64/asm.h

```diff
@@ -0,0 +1,85 @@
+#ifndef ASM_H
+#define ASM_H
+
+#ifdef __sgi
+#define _MIPS_ISA_MIPS1 1
+#define _MIPS_ISA_MIPS2 2
+#define _MIPS_ISA_MIPS3 3
+#define _MIPS_ISA_MIPS4 4
+#endif
+
+#ifndef _LANGUAGE_C
+
+#define LEAF(x)                 \
+    .balign 4                  ;\
+    .globl x                   ;\
+    .type x, @function         ;\
+    x:                         ;\
+        .ent x, 0              ;\
+        .frame $sp, 0, $ra
+
+#define XLEAF(x)                \
+    .balign 4                  ;\
+    .globl x                   ;\
+    .type x, @function         ;\
+    x:                         ;\
+        .aent x, 0
+
+#define NESTED(x, fsize, ra)    \
+    .globl x                   ;\
+    x:                         ;\
+        .ent x, 0              ;\
+        .frame $sp, fsize, ra
+
+#define XNESTED(x)              \
+    .globl x                   ;\
+    x:                         ;\
+        .aent x, 0
+
+#define END(x)                  \
```

Comment:
It is somewhat unfortunate but I think this is overall better. Without these the situation is really quite bad, you can't objdump the asm files properly which is now more worthwhile to do occasionally since they're now less literal.